src/runtime/importhook.cs

```diff
@@ -134,199 +130,135 @@ internal static void SaveRuntimeData(RuntimeDataStorage storage)
-            {
-                // Most likely a missing relative import.
-                // For example site-packages\bs4\builder\__init__.py uses it to check if a package exists:
-                //     from . import _html5lib
-                // We don't support them anyway
-                return IntPtr.Zero;
-            }
-            // Save the exception
-            var originalException = new PythonException();
-            // Otherwise,  just clear the it.
-            Exceptions.Clear();
 
-            string[] names = realname.Split('.');
+        /// <summary>
+        /// Because we use a proxy module for the clr module, we somtimes need
+        /// to force the py_clr_module to sync with the actual clr module's dict.
+        /// </summary>
+        internal static void UpdateCLRModuleDict()
+        {
+            root.InitializePreload();
 
-            // See if sys.modules for this interpreter already has the
-            // requested module. If so, just return the existing module.
-            IntPtr modules = Runtime.PyImport_GetModuleDict();
-            IntPtr module = Runtime.PyDict_GetItem(modules, py_mod_name);
+            // update the module dictionary with the contents of the root dictionary
+            root.LoadNames();
+            IntPtr py_mod_dict = Runtime.PyModule_GetDict(py_clr_module);
+            IntPtr clr_dict = Runtime._PyObject_GetDictPtr(root.pyHandle); // PyObject**
+            clr_dict = (IntPtr)Marshal.PtrToStructure(clr_dict, typeof(IntPtr));
+            Runtime.PyDict_Update(py_mod_dict, clr_dict);
+        }
 
-            if (module != IntPtr.Zero)
-            {
-                if (fromlist)
-                {
-                    if (IsLoadAll(fromList))
-                    {
-                        var mod = ManagedType.GetManagedObject(module) as ModuleObject;
-                        mod?.LoadNames();
-                    }
-                    Runtime.XIncref(module);
-                    return module;
-                }
-                if (clr_prefix != null)
-                {
-                    return GetCLRModule(fromList);
-                }
-                module = Runtime.PyDict_GetItemString(modules, names[0]);
-                Runtime.XIncref(module);
-                return module;
-            }
-            Exceptions.Clear();
+        /// <summary>
+        /// Return the clr python module (new reference)
+        /// </summary>
+        public static IntPtr GetCLRModule()
+        {
+
+            UpdateCLRModuleDict();
+            Runtime.XIncref(py_clr_module);
+            return py_clr_module;
+        }
 
-            // Traverse the qualified module name to get the named module
-            // and place references in sys.modules as we go. Note that if
+        /// <summary>
+        /// The hook to import a CLR module into Python
+        /// </summary>
+        public static ModuleObject __import__(string modname)
```

Comment:
This doesn't replace Python's `__import__` anymore so it should probably just be called `Import`